cms/src/main/java/uk/nhs/digital/apispecs/commonmark/CommonmarkMarkdownConverter.java

```diff
@@ -137,10 +137,10 @@ private String renderMarkdownText(final String markdown, final String headingIdP
     private String renderMarkdownModel(final Node document, final List<Extension> extensions, final String headingIdPrefix) {
 
         final HtmlRenderer renderer = HtmlRenderer.builder()
-            .nodeRendererFactory(context -> new FencedCodeBlockNodeRenderer(context))
-            .nodeRendererFactory(context -> new TableBlockNodeRenderer(context))
+            .nodeRendererFactory(FencedCodeBlockNodeRenderer::new)
+            .nodeRendererFactory(TableBlockNodeRenderer::new)
+            .nodeRendererFactory(context -> new HeadingsRenderer(context, headingIdPrefix))
```

Comment:
super pedantic, but could this class be renamed `HeadingNodeRenderer` to match it's siblings?

Reply:
also i like the use of the method reference operator for `FencedCodeBlockNodeRenderer` and `TableBlockNodeRenderer` :grin: